pkg/kubelet/cm/helpers_linux.go

```diff
@@ -141,7 +141,10 @@ func ResourceConfigForPod(pod *v1.Pod, enforceCPULimits bool, cpuPeriod uint64)
 	cpuLimitsDeclared := true
 	// map hugepage pagesize (bytes) to limits (bytes)
 	hugePageLimits := map[int64]int64{}
-	for _, container := range pod.Spec.Containers {
+	allContainers := []v1.Container{}
+	allContainers = append(allContainers, pod.Spec.Containers...)
+	allContainers = append(allContainers, pod.Spec.InitContainers...)
+	for _, container := range allContainers {
```

Comment:
I don't think the hugepage logic below is correct anymore.  Similar to the logic here: https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/api/v1/resource/helpers.go#L48, we probably want to sum the regular containers, and then take the max with init containers.